src/riak_ql_parser.yrl

```diff
@@ -935,26 +935,25 @@ canonicalise_expr({expr, X}) ->
 canonicalise_expr(X) ->
     X.
 
-get_func_type(FuncName) when FuncName =:= 'AVG'    orelse
-                             FuncName =:= 'MEAN'   orelse
-                             FuncName =:= 'SUM'    orelse
-                             FuncName =:= 'COUNT'  orelse
-                             FuncName =:= 'MIN'    orelse
-                             FuncName =:= 'MAX'    orelse
-                             FuncName =:= 'STDDEV' orelse
-                             FuncName =:= 'STDDEV_SAMP' orelse
-                             FuncName =:= 'STDDEV_POP' ->
-    window_aggregate_fn;
-get_func_type(FuncName) when is_atom(FuncName) ->
-    not_supported.
-
-%% TODO
-%% this list to atom needs to change to list to existing atom
-%% once the fns that actually execute the Window_Aggregates Fns are written then the atoms
-%% will definetely be existing - but just not now
-%% also try/catch round it
-canonicalise_window_aggregate_fn(Fn) when is_binary(Fn)->
-     list_to_atom(string:to_upper(binary_to_list(Fn))).
+get_func_type(FuncName) ->
+    Types = [Type || {Type, Mod} <- [{window_agg_fn, riak_ql_window_agg_fns},
+                                     {inverse_distrib_fn, riak_ql_inverse_distrib_fns}],
+                     lists:member(FuncName, Mod:supported_functions())],
+    case Types of
+        [Type] ->
+            Type;
+        [] ->
+            not_supported
+    end.
+
+%%
+canonicalise_fn(Fn) when is_binary(Fn)->
+    try
+        list_to_atom(string:to_upper(binary_to_list(Fn)))
```

Comment:
I *meant* it to be `list_to_existing_atom` (hence the `try`). Why doesn't emacs autocorrect such obvious typos?

Fixed in 1cc8aef6.